legend-engine-pure-code-compiled-core/src/main/resources/core/pure/binding/externalFormat/externalFormatContract.pure

```diff
@@ -0,0 +1,111 @@
+
+import meta::pure::generation::metamodel::*;
+
+import meta::pure::graphFetch::*;
+
+import meta::pure::metamodel::path::*;
+
+import meta::pure::model::unit::*;
+
+Class meta::external::shared::format::ExternalFormatContract<T>
+[
+  externalFormatMetamodelType : checkSuperType($this.externalFormatMetamodel, SchemaDetail)
+]
+{
+   id                               : String[1];   
+   contentTypes                     : String[*];
+   
+   externalFormatMetamodel          : Class<T>[1];
+
+   externalFormatToPureDescriptor   : ExternalFormatToPureDescriptor<SchemaToModelConfiguration>[0..1];
+   externalFormatFromPureDescriptor : ExternalFormatFromPureDescriptor<ModelToSchemaConfiguration>[0..1];
+   externalFormatBindingValidator   : Function<{Binding[1] -> BindingDetail[1]}>[0..1];
+
+   externalizeConfig                : Class<ExternalFormatExternalizeConfig>[0..1];
+   internalizeConfig                : Class<ExternalFormatInternalizeConfig>[0..1];
+   
+   externalFormatPlanDescriptor     : ExternalFormatPlanDescriptor[0..1];
+}
+
+Class meta::external::shared::format::ExternalFormatToPureDescriptor<U>
+[
+  configurationType : checkSuperType($this.configuration, SchemaToModelConfiguration)
+]
+{
+  configuration        : Class<U>[1];
+  generator            : Function<{SchemaSet[1], U[1] -> PackageableElement[*]}>[1];
+  generatorWithDebug   : Function<{SchemaSet[1], U[1] -> PackageableElement[*]}>[0..1];
+  defaultConfig        : U[1];
+}
+
+Class meta::external::shared::format::ExternalFormatFromPureDescriptor<V>
+[
+  configurationType : checkSuperType($this.configuration, ModelToSchemaConfiguration)
+]
+{
+  configuration        : Class<V>[1];
+  generator            : Function<{ModelUnit[1], V[1] -> SchemaSet[1]}>[1];
+  generatorWithDebug   : Function<{ModelUnit[1], V[1] -> SchemaSet[1]}>[0..1];
+  defaultConfig        : V[1];
+}
+
+Class <<typemodifiers.abstract>> meta::external::shared::format::ExternalFormatExternalizeConfig
+{
+
+}
+
+Class <<typemodifiers.abstract>> meta::external::shared::format::ExternalFormatInternalizeConfig
+{
+
+}
+
+Class meta::external::shared::format::executionPlan::ExternalFormatPlanDescriptor
+{ 
+  sourceRecordTree           : RootGraphFetchTree<Any>[1];
+  platformBindingDescriptors : ExternalFormatPlatformBindingDescriptor[*];
+}
+
+Class meta::external::shared::format::executionPlan::engine::ExternalFormatPlatformBindingDescriptor
+{
+   externalizeDescriptor  : ExternalFormatExternalizeBindingDescriptor[0..1];
+   internalizeDescriptor  : ExternalFormatInternalizeBindingDescriptor[0..1];
+}
+
+Class meta::external::shared::format::executionPlan::engine::ExternalFormatExternalizeBindingDescriptor
+{
+   externalizePreparator  : Function<{ExternalFormatExternalizeExecutionNode[1], String[1], meta::pure::executionPlan::engine::java::GenerationContext[1], meta::pure::extension::Extension[*], DebugContext[1] -> meta::pure::executionPlan::engine::java::typeInfo::TypeInfoSet[1]}>[0..1];

```

Comment:
Should we abstract out GenerationContext/Code to generic platform or we want to do that when we try with platform other than java?

Reply:
I think we should abstract out GenerationContext/TypeInfoSet now as it will help in moving Java specifics to a different repo. In the current state, it seems we are adding more dependencies to 'java' package classes which we want to avoid.

Reply:
To be addressed in following commits